src/test/java/net/snowflake/client/jdbc/ResultSetAsyncIT.java

```diff
@@ -169,4 +173,139 @@ public void testisClosedIsLastisAfterLast() throws SQLException {
+    String prepInsertString =
+        "insert into test_get values(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+    int bigInt = Integer.MAX_VALUE;
+    long bigLong = Long.MAX_VALUE;
+    short bigShort = Short.MAX_VALUE;
+    String str = "hello";
+    double bigDouble = Double.MAX_VALUE;
+    float bigFloat = Float.MAX_VALUE;
+    byte[] bytes = {(byte) 0xAB, (byte) 0xCD, (byte) 0x12};
+    BigDecimal bigDecimal = new BigDecimal("10000000000");
+    byte oneByte = (byte) 1;
+    Date date = new Date(44);
+    Time time = new Time(500);
+    Timestamp ts = new Timestamp(333);
+
+    Connection connection = getConnection();
+    Clob clob = connection.createClob();
+    clob.setString(1, "hello world");
+    Statement statement = connection.createStatement();
+    statement.execute(
+        "create or replace table test_get(colA integer, colB number, colC number, colD string, colE double, colF float, colG boolean, colH text, colI binary(3), colJ number(38,9), colK int, colL date, colM time, colN timestamp_ltz)");
+
+    PreparedStatement prepStatement = connection.prepareStatement(prepInsertString);
+    prepStatement.setInt(1, bigInt);
+    prepStatement.setLong(2, bigLong);
+    prepStatement.setLong(3, bigShort);
+    prepStatement.setString(4, str);
+    prepStatement.setDouble(5, bigDouble);
+    prepStatement.setFloat(6, bigFloat);
+    prepStatement.setBoolean(7, true);
+    prepStatement.setClob(8, clob);
+    prepStatement.setBytes(9, bytes);
+    prepStatement.setBigDecimal(10, bigDecimal);
+    prepStatement.setByte(11, oneByte);
+    prepStatement.setDate(12, date);
+    prepStatement.setTime(13, time);
+    prepStatement.setTimestamp(14, ts);
+    prepStatement.execute();
+
+    ResultSet resultSet =
+        statement.unwrap(SnowflakeStatement.class).executeAsyncQuery("select * from test_get");
+    resultSet.next();
+    assertEquals(bigInt, resultSet.getInt(1));
+    assertEquals(bigInt, resultSet.getInt("COLA"));
+    assertEquals(bigLong, resultSet.getLong(2));
+    assertEquals(bigLong, resultSet.getLong("COLB"));
+    assertEquals(bigShort, resultSet.getShort(3));
+    assertEquals(bigShort, resultSet.getShort("COLC"));
+    assertEquals(str, resultSet.getString(4));
+    assertEquals(str, resultSet.getString("COLD"));
+    Reader reader = resultSet.getCharacterStream("COLD");
+    char[] sample = new char[str.length()];
+
+    assertEquals(str.length(), reader.read(sample));
+    assertEquals(str.charAt(0), sample[0]);
+    assertEquals(str, new String(sample));
+
+    assertEquals(bigDouble, resultSet.getDouble(5), 0);
+    assertEquals(bigDouble, resultSet.getDouble("COLE"), 0);
+    assertEquals(bigFloat, resultSet.getFloat(6), 0);
+    assertEquals(bigFloat, resultSet.getFloat("COLF"), 0);
+    assertTrue(resultSet.getBoolean(7));
+    assertTrue(resultSet.getBoolean("COLG"));
+    assertEquals("hello world", resultSet.getClob("COLH").toString());
+
+    // TODO: figure out why getBytes returns an offset.
```

Comment:
Found a bug here with getBytes(). It returns a Byte value with an offset. Created a Jira to investigate this.